### PR TITLE
Machines destroy command

### PR DESCRIPTION
**Pull request by jeffandersen**
```
> torus machines list

ID                                   NAME                              STATE         TEAM                     CREATION DATE

04buv5v36xecb5r390btvevkh0k10        api-test-machines-00cjftz6        active        api-test-machines        2016-10-30T12:56:21Z
04bh45uqk811xqzr81jb6naw5mt02        api-test-machines-b5hryd28        active        api-test-machines        2016-10-30T12:56:35Z
```
```
> torus machines destroy api-test-machines-b5hryd28
You are about to destroy a machine. This cannot be undone.
✗ Do you wish to continue? [Y/n] n
```
```
> torus machines destroy api-test-machines-b5hryd28
You are about to destroy a machine. This cannot be undone.
✔ Do you wish to continue? [Y/n] y
Machine destroyed.
```
```
> torus machines list --destroyed

ID                                   NAME                              STATE            TEAM                     CREATION DATE

04bh45uqk811xqzr81jb6naw5mt02        api-test-machines-b5hryd28        destroyed        api-test-machines        2016-10-30T12:56:35Z
```

```
> torus machines destroy 04buv5v36xecb5r390btvevkh0k10 -y
Machine destroyed.
```

### CODE REVIEW

File: cmd/machines.go [hunk 1]
```diff
@@ -58,6 +58,19 @@ func init() {
 				),
 			},
 			{
+				Name:      "destroy",
+				Usage:     "Destroy a machine in the specified organization",
+				ArgsUsage: "<identity>",
```
Comment (ianlivingstone) [inline cmd/machines.go]: we should specify the identity as a machine id or machine name (not sure how its done with `machines view` but `<identity>` doesn't really tell me "what an identity is"

File: cmd/machines.go [hunk 2]
```diff
@@ -75,6 +88,64 @@ func init() {
...
+		machineID = *machines[0].Machine.ID
+	}
+
+	preamble := "You are about to destroy a machine. This cannot be undone."
+	abortErr := ConfirmDialogue(ctx, nil, &preamble)
+	if abortErr != nil {
+		return abortErr
+	}
+
+	// TODO: Get should take an orgID and send as query param
```
Comment (ianlivingstone) [inline cmd/machines.go]: Forgotten comment?

File: cmd/machines.go [hunk 3]
```diff
@@ -75,6 +88,64 @@ func init() {
 	Cmds = append(Cmds, machines)
 }
 
+func destroyMachineCmd(ctx *cli.Context) error {
+	args := ctx.Args()
+	if len(args) > 1 {
+		return errs.NewUsageExitError("Too many arguments supplied.", ctx)
+	}
+	if len(args) < 1 {
+		return errs.NewUsageExitError("Identity is required", ctx)
```
Comment (ianlivingstone) [inline cmd/machines.go]: arg name change (see comment in help section)

File: cmd/machines.go (hunk 2 above)
Comment (jeffandersen) [inline cmd/machines.go]: This is a todo that the registry should require it like the other endpoints.

**Pull request merged**